pkg/core/mempool/mempool.go

```diff
@@ -145,7 +143,7 @@ func (m *Mempool) Run() {
 			case <-time.After(20 * time.Second):
 				m.onIdle()
 			// Mempool terminating
-			case <-m.quitChan:
+			case <-m.ctx.Done():
```

Comment:
a better option than saving the context in the `Mempool` state would be to introduce `context.Context` as parameter of `Run` and let the server call it instead of implicitly being called by the constructor.

Reply:
Yep, makes sense!